classes/Product.php

```diff
@@ -764,10 +762,13 @@ public function __construct($id_product = null, $full = false, $id_lang = null,
     public function getFieldsShop()
     {
         $fields = parent::getFieldsShop();
-        if (null === $this->update_fields || (!empty($this->update_fields['price']) && !empty($this->update_fields['unit_price']))) {
-            if ($this->unit_price !== null) {
-                $fields['unit_price_ratio'] = (float) $this->unit_price > 0 ? $this->price / $this->unit_price : 0;
-            }
+
+        // In case unit_price is not in fields we can try and fallback with unit_price_ratio
+        if (null !== $this->unit_price_ratio && empty($fields['unit_price']) && (
```

Comment:
Shouldn't ```empty($fields['unit_price'])``` be the first criteria ?

Reply:
I'm not sure both make sense and I don't think the order is important here anyway